synapse/storage/databases/main/devices.py

```diff
@@ -221,6 +222,8 @@ async def get_device_updates_by_remote(
             limit,
         )
 
+        # len(updates) <= limit.
```

Comment:
What's going on here? Oh, you're tracking an upper bound on the size of `updates`.

Maybe this?

```suggestion
        # We need to ensure `updates` doesn't grow too big. 
        # Currently: `len(updates) <= limit`.
```